### PR TITLE
Faster align

### PR DESCRIPTION
**Pull request by jmiller656**
Improve alignment speed by using biojava's structureIO API, which caches structures locally

Do not merge until #125 is merged

### CODE REVIEW

File: src/main/java/org/moltimate/moltimatebackend/util/StructureUtils.java [hunk 1]
```diff
@@ -195,24 +197,18 @@ private static double rmsd(double[] point1, double[] point2) {
...
-                    .findFirst()
-                    .get()
-                    .getEcNums()
-                    .stream()
-                    .filter(Objects::nonNull)
-                    .findFirst()
-                    .get();
-        } catch (Exception e) {
-            return "unknown";
+        RCSBDescription description = RCSBDescriptionFactory.get(structure.getPDBCode());
```
Comment (blackpan2) [inline src/main/java/org/moltimate/moltimatebackend/util/StructureUtils.java]: 🔥

**Comment by blackpan2** [comment]
This caches structures instead of always getting them from the PDB. This would only help us in the case of repeatedly getting a structure from the PDB, which we are caching?

I like the other changes just checking in

**Comment by jmiller656** [comment]
It essentially saves us ~1000 network calls per alignment, which helps for a ton of reasons. Keeps the structures locally

**Comment by blackpan2** [comment]
But wait why is it saving us so many calls? Where are we getting the structure multiple times?

**Comment by jmiller656** [comment]
It happens every time we call `ProteinUtils.queryPDB`. This gets called for every single motif every time when we do alignments: https://github.com/moltimate/moltimate-backend/blob/40156b89489310e281eeb98349d71bad736b4439/src/main/java/org/moltimate/moltimatebackend/service/AlignmentService.java#L143

**Comment by blackpan2** [comment]
Ok yes, I see that now. Do we have any way to do settings on the caching for structures that this does? We reference a lot of different proteins and this could drive memory usage way up.

File: src/main/java/org/moltimate/moltimatebackend/util/StructureUtils.java [hunk 2]
```diff
@@ -195,24 +197,18 @@ private static double rmsd(double[] point1, double[] point2) {
...
+        for(RCSBPolymer polymer: description.getPolymers()){
+            if(polymer.getEnzClass() != null){
+                return polymer.getEnzClass();
+            }
         }
+        return "unknown";
```
Comment (steplica) [inline src/main/java/org/moltimate/moltimatebackend/util/StructureUtils.java]: Can you make this a public final constant value so we can check if `ecNumber(...) == Constants.UNKNOWN_EC_NUMBER`

**Comment by jmiller656** [comment]
> 
> 
> Ok yes, I see that now. Do we have any way to do settings on the caching for structures that this does? We reference a lot of different proteins and this could drive memory usage way up.

According to [biojava](https://github.com/biojava/biojava-tutorial/blob/master/structure/caching.md), the cache is stored in the local filesystem. So it does not use memory

**Pull request merged**